dot/state/block.go

```diff
@@ -479,6 +480,26 @@ func (bs *BlockState) SetBlockBody(hash common.Hash, body *types.Body) error {
 	return bs.db.Put(blockBodyKey(hash), encodedBody)
 }
 
+// SetFirstNonOriginSlotNumber saves the first non-origin slot number into the DB
+func (bs *BlockState) SetFirstNonOriginSlotNumber(hash common.Hash, slotNumber uint64) error {
+	buf := make([]byte, 8)
+	binary.LittleEndian.PutUint64(buf, slotNumber)
+	return bs.db.Put(firstSlotNumberKey, buf)
+}
+
+// GetFirstNonOriginSlotNumber returns the slot number of the first non origin block
+func (s *BlockState) GetFirstNonOriginSlotNumber() (uint64, error) {
```

Comment:
Same with this, do you want to export it? it is only used within the same package.